package/MDAnalysis/auxiliary/XVG.py

```diff
@@ -76,21 +76,35 @@ def _read_next_step(self):
 
 
     def count_n_steps(self):
-        """ Iterate through all steps to count total number and make times list.
+        """ Iterate through all steps to count total number.
 
         Returns
         -------
         int
-           Total number of steps
+            Total number of steps
+        """
+
+        if self.constant_dt:
+            # will have to iterate through all to built times list anyway
+            return len(self.read_all_times())
+        else:
+            # iterate here instead
+            self._restart()
+            count = 0
+            for step in self:
+                count = count + 1
+            return count
+
+    def read_all_times(self):

```

Comment:
But you do not always assume contant *dt* (which I like). So I guess I guess it makes sense assuming the number of records is not too large.